code/modules/projectiles/guns/energy/capacitor.dm

```diff
@@ -0,0 +1,163 @@
+/obj/item/gun/energy/capacitor
+	name = "capacitor pistol"
+	desc = "An excitingly chunky directed energy weapon that uses a modular capacitor array to charge each shot."
+	icon_state = "world"
+	on_mob_icon = 'icons/obj/guns/capacitor_pistol.dmi'
+	icon = 'icons/obj/guns/capacitor_pistol.dmi'
+	icon_state = "world"
+	origin_tech = "{'combat':4,'materials':4,'powerstorage':4}"
+	w_class = ITEM_SIZE_NORMAL
+	projectile_type = /obj/item/projectile/beam/midlaser
+	charge_cost = 25
+	accuracy = 2
+	fire_delay = 10
+	slot_flags = SLOT_BELT
+	cell_type = /obj/item/cell/high
+	material = /decl/material/solid/metal/steel
+	matter = list(
+		/decl/material/solid/glass = MATTER_AMOUNT_REINFORCEMENT,
+		/decl/material/solid/gemstone/diamond = MATTER_AMOUNT_TRACE
+	)
+
+	var/capacitor_type = /obj/item/stock_parts/capacitor
+	var/capacitor_charge_constant = 10
+	var/list/capacitors
+	var/max_capacitors = 2
+	var/charging = FALSE
+	var/charge_damage_multiplier = 0.5
+
+/obj/item/gun/energy/capacitor/Destroy()
+	if(capacitors)
+		QDEL_NULL_LIST(capacitors)
+	. = ..()
+
+/obj/item/gun/energy/capacitor/Initialize()
+	var/capacitor_volume = 0
+	for(var/i = 1 to max_capacitors)
+		var/obj/item/stock_parts/capacitor/capacitor = new capacitor_type(src)
+		capacitor_volume = min(capacitor_volume, capacitor.max_charge)
+		LAZYADD(capacitors, capacitor)
+	. = ..()
+
+/obj/item/gun/energy/capacitor/afterattack(atom/A, mob/living/user, adjacent, params)
+	. = !charging && ..()
+	
+/obj/item/gun/energy/capacitor/attackby(obj/item/W, mob/user)
+	if(!charging)
+
+		if(isScrewdriver(W))
+			if(length(capacitors))
+				var/obj/item/capacitor = capacitors[1]
+				capacitor.dropInto(user.loc)
+				user.put_in_hands(capacitor)
```

Comment:
Dropping into `user.loc` is redundant with `put_in_hands`, as that's the fail behavior anyway. Unsure if you meant `loc` for action-at-a-distance stuff or if it should be dropped.